### PR TITLE
New e2e

### PR DESCRIPTION
**Pull request by khassel**
Related to #2408

Managed to move more tests using `electron` to use `server only`.

Because this new tests are using `jsdom` and therefore running in the browser all test configs used must be changed because `require` cannot be used anymore.

There are 2 module tests still using electron (calendar and weather). They are complex (using basic-auth or mock-data), will look into this in a following PR.

Beside the test stuff only `logger.js` and `server.js` are changed and I tested this in `server only` and with `electron` on a pi3.

### CODE REVIEW

File: tests/configs/modules/alert/default.js
Comment (rejas) [inline tests/configs/modules/alert/default.js:7]: Do these config infos need to be in each file? Cant they be refactored into a common file?

Comment (khassel) [reply tests/configs/modules/alert/default.js:7]: tell me how, `require` cannot be used in the browser ...

Comment (khassel) [reply tests/configs/modules/alert/default.js:7]: Solution is simple, we can remove all the default values which are defined in `js/default.js`. Will add a commit.

**Comment by khassel** [comment]
@MichMich the upload error to Codecov seems unrelated to this PR

File: tests/configs/without_modules.js
Comment (rejas) [inline tests/configs/without_modules.js:9]: do we need that unit config entry?

**Review by rejas — APPROVED** [review]

File: tests/configs/without_modules.js
Comment (khassel) [reply tests/configs/without_modules.js:9]: no, did an automatic search/replace, will check again and update.

**Comment by rejas** [comment]
Looks good to me and though it doesnt change much in the non-test code, maybe we wait until the october release is out to merge it? Juuuust to be on the safe side :-D

File: tests/configs/modules/alert/default.js
Comment (rejas) [reply tests/configs/modules/alert/default.js]: Yeah, I thought about using import but then a lot more would have to be refactored IIRC

**Comment by khassel** [comment]
> maybe we wait until the october release

let we decide @MichMich , have no problem to wait ...

**Comment by MichMich** [comment]
> Looks good to me and though it doesnt change much in the non-test code, maybe we wait until the october release is out to merge it? Juuuust to be on the safe side :-D

Whoops! Missed that one. Well, let's start living on the edge. 😂

**Pull request merged**
